server/src/Server.js

```diff
@@ -0,0 +1,119 @@
+import http from 'http';
+import express from 'express';
+import bodyParser from 'body-parser';
+import crypto from 'crypto';
+import {MongoClient} from 'mongodb';
+import assert from 'assert';
+
+const app = new express();
+const server = http.createServer(app);
+
+const DB_ADDRESS = 'mongodb://localhost:27017/car-app';
+let db = null;
+app.use(bodyParser.json());
+
+app.post('/checkUsername', (req, res) => {
+
+    const username = req.body.username.toLowerCase();
+    const collection = db.collection('login');
+
+    console.log(`Checking username ${username}`);
+
+    collection.find({ username: username}).toArray((err, result) => {
+        if (result.length > 0) {
+            console.log(`${username} exists.`);
+            res.json({ exists: true});
+        } else {
+            console.log(`${username} not found.`);
+            res.json({ exists: false});
+        }
+    });
+});
+
+app.post('/login', (req, res) => {
+    const collection = db.collection('login');
+    const login = {
+        username: req.body.username.toLowerCase(),
+        password: req.body.password,
+    };
+
+    console.log('Checking Login');
+
+    collection.find({ username: login.username}).toArray((err, result) => {
+        if (err) throw err;
+        if (result.length !== 1) {
+            throw console.log('Error in username number!' + result);
+        }
+
+        const loginAttempt = result[0];
+
+        if (verifyLogin(loginAttempt, login.password)) {
+            res.sendStatus(200).end();
+            console.log(`${login.username} login verified.`);
+        } else {
+            res.sendStatus(401).end();
+            console.log('Bad Login');
+        }
+    });
+});
+
+app.post('/register', (req, res) => {
+    const login = {
+        username: req.body.username.toLowerCase(),
+        password: req.body.password,
+    };
+
+    console.log(`registering ${login.username}`);
+    const ITERATION = 20000;
+    const salt = crypto.randomBytes(16);
+    const hashed = crypto.pbkdf2Sync(login.password, salt, ITERATION, 256, 'sha256').toString('hex');
```

Comment:
Consider using the async version of this method.  Under a heavy server load, the sync version will prevent simultaneous clients from getting shit done.